src/scheduler/server_info.cpp

```diff
@@ -2107,156 +2072,131 @@ check_resv_running_on_node(resource_resv *resv, const void *arg)
+	if (osinfo.fstree != NULL)
+		fstree = new fairshare_head(*osinfo.fstree);
+	has_mult_express = osinfo.has_mult_express;
+	has_soft_limit = osinfo.has_soft_limit;
+	has_hard_limit = osinfo.has_hard_limit;
+	has_user_limit = osinfo.has_user_limit;
+	has_all_limit = osinfo.has_all_limit;
+	has_grp_limit = osinfo.has_grp_limit;
+	has_proj_limit = osinfo.has_proj_limit;
+	has_multi_vnode = osinfo.has_multi_vnode;
+	has_prime_queue = osinfo.has_prime_queue;
+	has_nonprime_queue = osinfo.has_nonprime_queue;
+	has_ded_queue = osinfo.has_ded_queue;
+	has_nodes_assoc_queue = osinfo.has_nodes_assoc_queue;
+	node_group_enable = osinfo.node_group_enable;
+	eligible_time_enable = osinfo.eligible_time_enable;
+	provision_enable = osinfo.provision_enable;
+	power_provisioning = osinfo.power_provisioning;
+	use_hard_duration = osinfo.use_hard_duration;
+	pset_metadata_stale = osinfo.pset_metadata_stale;
+	name = osinfo.name;
+	liminfo = lim_dup_liminfo(osinfo.liminfo);
+	server_time = osinfo.server_time;
+	res = dup_resource_list(osinfo.res);
+	alljobcounts = dup_counts_map(osinfo.alljobcounts);
+	group_counts = dup_counts_map(osinfo.group_counts);
+	project_counts = dup_counts_map(osinfo.project_counts);
+	user_counts = dup_counts_map(osinfo.user_counts);
+	total_alljobcounts = dup_counts_map(osinfo.total_alljobcounts);
+	total_group_counts = dup_counts_map(osinfo.total_group_counts);
+	total_project_counts = dup_counts_map(osinfo.total_project_counts);
+	total_user_counts = dup_counts_map(osinfo.total_user_counts);
+	node_group_key = osinfo.node_group_key;
+	nodesigs = dup_string_arr(osinfo.nodesigs);
+
+	policy = dup_status(osinfo.policy);
+
+	num_nodes = osinfo.num_nodes;
 
 	/* dup the nodes, if there are any nodes */
-	nsinfo->nodes = dup_nodes(osinfo->nodes, nsinfo, NO_FLAGS);
+	nodes = dup_nodes(osinfo.nodes, this, NO_FLAGS);
 
-	if (nsinfo->has_nodes_assoc_queue) {
-		nsinfo->unassoc_nodes =
-			node_filter(nsinfo->nodes, nsinfo->num_nodes, is_unassoc_node, NULL, 0);
+	if (has_nodes_assoc_queue) {
+		unassoc_nodes =
+			node_filter(nodes, num_nodes, is_unassoc_node, NULL, 0);
 	} else
-		nsinfo->unassoc_nodes = nsinfo->nodes;
+		unassoc_nodes = nodes;
 
-	nsinfo->unordered_nodes = dup_unordered_nodes(osinfo->unordered_nodes, nsinfo->nodes);
+	unordered_nodes = dup_unordered_nodes(osinfo.unordered_nodes, nodes);
 
 	/* dup the reservations */
-	nsinfo->resvs = dup_resource_resv_array(osinfo->resvs, nsinfo, NULL);
-	nsinfo->num_resvs = osinfo->num_resvs;
+	resvs = dup_resource_resv_array(osinfo.resvs, this, NULL);
+	num_resvs = osinfo.num_resvs;
 
 #ifdef NAS /* localmod 053 */
 	site_save_users();
 #endif /* localmod 053 */
 
 	/* duplicate the queues */
-	nsinfo->queues = dup_queues(osinfo->queues, nsinfo);
-	if (nsinfo->queues.empty()) {
-		free_server(nsinfo);
-		return NULL;
+	queues = dup_queues(osinfo.queues, this);
+	if (queues.empty()) {
+		free_server_info();
+		throw sched_exception("Unable to duplicate queues", SCHD_ERROR);
```

Comment:
If the int in sched_exception is a sched error code, then it shouldn't be int.  It should be the enum.